### DISCUSSION TITLE
how can I separate chart values into categories that contain more than one value?

### DISCUSSION THREAD

**Discussion by nextersolutions** [patrykandpatrick/vico#808] [Questions]
### Question

![Graphs](https://github.com/user-attachments/assets/08340df4-b822-4668-bb4a-ade26fc746a5)

I need to make a chart look as shown above and, according to docs, I can add the categories like this (GraphCalculationUtil is my tool to segment data and convert a values array into the linked map):
```
val data = GraphCalculationUtil.calculateAxisLabel(values, selectedRange)
val xToDateMapKey = ExtraStore.Key<Map<Float, String>>()
val xToDates = data.keys.associateBy { it.hashCode().toFloat() }
```

Then I add it like this:
```
lineSeries { series(xToDates.keys, values)  }
updateExtras { it[xToDateMapKey] = xToDates }
```

But it doesn't work. The graph area displays nothing. I think it's because while segmentation I provide X values only for the segments, I mean, where should be placed the vertical line and omit remaining items and don't provide X values for the remaining. I only indicate indexes, where segmenting lines should be placed, it's a logical approach.

My valueFomatter looks this way:
```
valueFormatter = { x, chartValues, _ ->
    try {
       chartValues.model.extraStore[xToDateMapKey][x] ?: ""
    } catch (e: Exception) { "" }
 }

```
The thing is, in [docs](https://patrykandpatrick.com/vico/wiki/cartesian-charts/layers/column-layer) and in the [example1](https://github.com/patrykandpatrick/vico/blob/v2.0.0-alpha.22/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart2.kt) and [example2](https://github.com/patrykandpatrick/vico/blob/v2.0.0-alpha.22/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart6.kt) there are examples shown only for categories, where the X value for present for each Y value, which doesn't solve my issue of multiple items categorization.

I cannot use one record per category, so my question is: how to add more than 1 record to a single category? Please, help

### Vico version(s)

2.0.0-alpha.19

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @ordinarythinker.

> according to docs, I can add the categories like this

This doesn't match what's suggested in the documentation. You should use timestamps, not hash codes, for the _x_ values. Hash codes don't represent anything that is of relevance here. By using them, you're creating a list of _x_ values that effectively has nothing to do with your data.

> But it doesn't work. The graph area displays nothing

We'd need a complete, runnable snippet to definitively determine what's wrong, but it's likely the above, the following, or both.

> I only indicate indexes, where segmenting lines should be placed, it's a logical approach.

It would be logical for `ItemPlacer.getLineValues`, but it's not for the series, which should include the data for the line. Where the `HorizontalAxis` displays labels and lines is a completely separate issue.

> there are examples shown only for categories, where the X value for present for each Y value

A _y_ value must have an _x_ value. This is the nature of the Cartesian coordinate system. What you're looking to do is not to add _y_ values with no _x_ values but rather to have the `HorizontalAxis` create segments in a particular way and not label every _x_ value individually.

> how to add more than 1 record to a single category?

Generally, this is achieved via `getXStep` and `ItemPlacer`. However, the chart in the attached image doesn't seem to make sense mathematically. The time between 10 AM on October 13 and 11 PM on October 14 is 37 hours. The time between 11 PM on October 14 and 3 PM on some following day can't possibly be 37 hours. The options are 16 hours, 40 hours, and so on. The axis isn't linear. Could you clarify how this is supposed to work?
